PROTOCOL.md

```diff
@@ -331,6 +347,39 @@ The following is an example `remove` action.
 }
 ```
 
+### Add CDC File
+The `cdc` action is used to add a [file](#change-data-files) containing only the data that was changed as part of the transaction. When CDC readers encounter a `cdc` action in a particular Delta table version, they must read the changes made in that version exclusively using the `cdc` files. If a version has no `cdc` action, then the data in `add` and `remove` actions are read as inserted and deleted rows, respectively.
+
+The schema of the `cdc` action is as follows:
+
+Field Name | Data Type | Description
+-|-|-
+path| String | A relative path to a change data file from the root of the table or an absolute path to a change data file that should be added to the table. The path is a URI as specified by [RFC 2396 URI Generic Syntax](https://www.ietf.org/rfc/rfc2396.txt), which needs to be decoded to get the file path.
+partitionValues| Map[String, String] | A map from partition column to value for this file. See also [Partition Value Serialization](#Partition-Value-Serialization)
+size| Long | The size of this file in bytes
+tags | Map[String, String] | Map containing metadata about this file
+
+The following is an example of `cdc` action.
+
+```
+{
+  "cdc": {
+    "path": "_change_data/cdc-00001-c…..snappy.parquet",
+    "partitionValues": {},
+    "size": 1213,
+    "dataChange": false
+  }
+}
+```
+
+#### Writer Requirements for AddCDCFile
+
+As of [Writer Version 4](#Writer-Version-Requirements), all writers must respect the `delta.enableChangeDataFeed` configuration flag in the metadata of the table. When `delta.enableChangeDataFeed` is `true`, writers must produce the relevant `AddCDCFile`'s for any operation that changes data, as specified in [Change Data Files](#change-data-files)
+
+#### Reader Requirements for AddCDCFile
+
+When available, change data readers should use the `AddCDCFile`s in a given table version instead of computing changes from the underlying data files referenced by the `add` and `remove` actions.
```

Comment:
also, this statement can be made more crisp. something like 
```
When reading the row-level changes made in a version, the following strategy should be used
- If there are AddCDCFile actions, then read them
- Otherwise read AddFile action (treat all rows as Inserts) and RemoveFile actions (treat all rows as deletes).